### PR TITLE
Correct Channel outbound HTLC serialization and expand fuzzing coverage

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This is based on #851 as it is required to get chanmon_consistency_target to run clean anyway.

This fixes a few trivial bugs in full_stack_target, then expands the coverage of chanmon_consistency_target significantly, fixing one bug that that expansion found (so far).

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Rebased after #851 merge.

File: fuzz/src/chanmon_consistency.rs
Comment (jkczyz) [inline fuzz/src/chanmon_consistency.rs:586]: After reading through the code, I understand how this is used. But I think a comment would helpful for posterity.

File: fuzz/src/full_stack.rs
Comment (TheBlueMatt) [reply fuzz/src/full_stack.rs:546]: We can broadcast transactions from inside `connect_block`, eg from the `ChannelMonitor`.

Comment (jkczyz) [reply fuzz/src/full_stack.rs:546]: Ah, for some reason I stopped reading `connect_block` at the end of the for loop. :P

**Comment by TheBlueMatt** [comment]
Added some additional documentation.

**Review by jkczyz — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed fixups with no changes. Will merge after CI:
```
$ git diff-tree -U3  1c818cd13 25dbd0d7e
$
```

**Pull request merged**
